src/main/java/io/quarkus/gizmo2/impl/BlockCreatorImpl.java

```diff
@@ -682,13 +682,13 @@ public Expr cast(final Expr a, final ClassDesc toType) {
 
     public Expr uncheckedCast(final Expr a, final ClassDesc toType) {
         if (a.type().isPrimitive()) {
-            throw new IllegalArgumentException("Only object types may be unsafely cast");
+            throw new IllegalArgumentException("Only object types may be unchecked cast");
```

Comment:
"unchecked" is an adjective, not an adverb, so you'd have to say something like "Unchecked cast may only apply to object types".